### PR TITLE
Fix collateral token behaviors on split position

### LINKED ISSUE #223: Add custom token: Error messages for the contract validations show code/no validation message about unsuccessful attempt of unlocking an account
UPD: 17/09/2020
Validation messages are not displayed at all not on the pop-uo, however, they are displayed in console
image

Related to #202

1. Error messages for the contract validations show code: see the video: 
https://drive.google.com/file/d/1UWUkgwc-cHCm_I1pQiU-4sijvPLwVBPK/view
**Expected result:** messages are written in understandable to an end user manner

2. No validation message about unsuccessful attempt of unlocking an account (see the video)
 https://drive.google.com/file/d/1kiGBF0UK4GaG6wDyXEEYXKZwF65DgOoT/view

**Expected result:** Error message should appear on the window

### PR DESCRIPTION
**Pull request by davidalbela**
- [x] Set `humanizeMessageError` on useCollateral hook.
- [x] Add message error into setAllowance when a tx is rejected.

Closes #223

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr331--conditionaltokensexplorer.review.gnosisdev.com

**Review by hpmaxi — APPROVED** [review]

File: src/components/common/SetAllowance/index.tsx [hunk 1]
```diff
@@ -51,16 +51,24 @@ interface Props {
...
+  error: string
   onUnlock: () => void
 }
 
 export const SetAllowance = (props: Props) => {
-  const { collateral, fetching, finished, onUnlock } = props
+  const { collateral, error, fetching, finished, onUnlock } = props
   const btnText = fetching ? 'Working...' : finished ? 'Done!' : 'Unlock'
 
+  console.log('ERROR: ' + error)
```
Comment (mariano-aguero) [inline src/components/common/SetAllowance/index.tsx:62]: Please use logger instead

File: src/components/common/SetAllowance/index.tsx [hunk 2]
```diff
@@ -51,16 +51,24 @@ interface Props {
   collateral: any
   fetching: boolean
   finished: boolean
+  error: string
```
Comment (mariano-aguero) [inline src/components/common/SetAllowance/index.tsx:54]: Also add a better type check for this, I think that collateral must be of type token instead, and possible error must be of type Error

File: src/hooks/useAllowanceState.tsx [hunk 3]
```diff
@@ -20,6 +20,7 @@ export const useAllowanceState = (allowanceMethods: AllowanceMethods, amount: Bi
   const [hasUnlockedCollateral, setHasUnlockedCollateral] = useState(false)
   const [allowanceFinished, setAllowanceFinished] = useState(false)
   const [shouldDisplayAllowance, setShouldDisplayAllowance] = useState(true)
+  const [allowanceError, setAllowanceError] = useState('')
```
Comment (mariano-aguero) [inline src/hooks/useAllowanceState.tsx:23]: This should be of type <Maybe<Error>> and the default must be _null_

File: src/hooks/useAllowanceState.tsx [hunk 4]
```diff
@@ -70,5 +72,6 @@ export const useAllowanceState = (allowanceMethods: AllowanceMethods, amount: Bi
     unlockCollateral,
     shouldDisplayAllowance,
     allowanceFinished,
+    allowanceError: allowanceError,
```
Comment (mariano-aguero) [inline src/hooks/useAllowanceState.tsx:75]: Not necessary to add the property, just the _allowanceError_ should be enough

File: src/util/tools.ts [hunk 5]
```diff
@@ -305,8 +305,8 @@ export const getMergePreview = (
   }
 }
 
-const humanizeMessageError = (error: string): string => {
-  let result = error
+export const humanizeMessageError = (error: string): string => {
```
Comment (mariano-aguero) [inline src/util/tools.ts:308]: Just an enhancement, the proper way to type a variable as a string and add a default value , all in one line, is in this way

```error = ''```

The compiler will understand that as a string, and you have a default value too

Just a link to explorer more on this ...https://www.typescripttutorial.net/typescript-tutorial/typescript-default-parameters/

**Comment by mariano-aguero** [comment]
Great work @davidalbela , to take into account, please fix the conflicts

File: src/util/tools.ts (hunk 5 above)
Comment (davidalbela) [reply src/util/tools.ts]: done!

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr331--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr331--conditionaltokensexplorer.review.gnosisdev.com

**Review by hpmaxi — APPROVED** [review]

File: src/components/form/SelectCollateral/index.tsx [hunk 6]
```diff
@@ -13,12 +13,14 @@ export interface SelectCollateralProps {
   formMethods: FormContextValues<SplitPositionFormMethods>
   splitFromCollateral: boolean
   tokens: Token[]
+  setAllowanceError: any
```
Comment (mariano-aguero) [inline src/components/form/SelectCollateral/index.tsx:16]: Add a proper type here

Comment (mariano-aguero) [reply src/components/form/SelectCollateral/index.tsx:16]: Mmmm I think with `Maybe<Error>` should be OK, please check it before commit

Comment (davidalbela) [reply src/components/form/SelectCollateral/index.tsx]: Replaced setAllowationError to `cleanAllowationError` callback

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr331--conditionaltokensexplorer.review.gnosisdev.com

**Review by mariano-aguero — APPROVED** [review]

**Pull request merged**